specklepy/serialization/base_object_serializer.py

```diff
@@ -266,20 +266,21 @@ def recompose_base(self, obj: dict) -> Base:
         object_type = Base.get_registered_type(speckle_type)
 
         # initialise the base object using `speckle_type` fall back to base if needed
-        base = object_type() if object_type else Base(speckle_type=speckle_type)
+        totalChildrenCount = 0
+        base_object_dict = {}
         # get total children count
         if "__closure" in obj:
             if not self.read_transport:
                 raise SpeckleException(
                     message="Cannot resolve reference - no read transport is defined"
                 )
             closure = obj.pop("__closure")
-            base.totalChildrenCount = len(closure)
+            totalChildrenCount = len(closure)
 
         for prop, value in obj.items():
             # 1. handle primitives (ints, floats, strings, and bools) or None
             if isinstance(value, PRIMITIVES) or value is None:
-                base.__setattr__(prop, value)
```

Comment:
this is what allows us to set both fields and `@property`s. if there is another way to do this or another way to initialise the object below that will also set properties, I would be keen to see it! however with the current changes, properties will not be set at all.